### ISSUE TITLE
Bug: traefik v2.6 panic in v0.4.0, v0.2.0 works

### ISSUE THREAD

**Issue by LordMike** [nscuro/traefik-plugin-geoblock#23]
I'm running the geoblock plugin v0.4, with Traefik v2.6. It panics at the below point. I've also run this in v2.5.

* Line 86 seems to be the first line that checks something with the config. So perhaps that's incorrect.. 
* I tried renaming the config variables to match the [names here](https://github.com/nscuro/traefik-plugin-geoblock/blob/559d3c116351b5cdcaa35a7c52b26a95c5fcd5db/plugin.go#L77-L80), but that didn't help. I'm not sure why the stack trace doesn't reference this plugin, _at all_..
* So I looked up the error message, and it has something to do with Go pointers - so I checked the history and did a checkout of `v0.2.0`, which is before the `p` variable changed from being access as a pointer, to just the variable itself.
* `v0.2.0` does not panic, and blocks my request as expected
```
# custom.yml, registered as a file provider
http:
  middlewares:
    https-redirect:
      redirectScheme:
        scheme: https
    geoblock:
      plugin:
        geoblock:
          enabled: true
          databaseFilePath: /plugins-local/src/github.com/nscuro/traefik-plugin-geoblock/IP2LOCATION-LITE-DB1.IPV6.BIN
          allowedCountries: ["GB"]
          #allowedCountries: ["DK","GB"]
          allowPrivate: true
          disallowedStatusCode: 403
```

```
time="2022-04-01T18:59:56Z" level=error msg="plugins-local/src/github.com/nscuro/traefik-plugin-geoblock/plugin.go:86:6: panic" module=github.com/nscuro/traefik-plugin-geoblock plugin=plugin-geoblock
time="2022-04-01T18:59:56Z" level=error msg="Recovered from panic in HTTP handler [80.210.71.237:34137 - /dashboard/projects/starred]: errors: *target must be interface or implement error" middlewareName=traefik-internal-recovery middlewareType=Recovery
time="2022-04-01T18:59:56Z" level=error msg="Stack: goroutine 218 [running]:
github.com/traefik/traefik/v2/pkg/middlewares/recovery.recoverFunc({0x3fa6148, 0xc001614a70}, 0xc0006c9700)
	github.com/traefik/traefik/v2/pkg/middlewares/recovery/recovery.go:46 +0x225
panic({0x2df71e0, 0x3f19740})
	runtime/panic.go:1038 +0x215
github.com/traefik/yaegi/interp.runCfg.func1()
	github.com/traefik/yaegi@v0.11.2/interp/run.go:193 +0x145
panic({0x2df71e0, 0x3f19740})
	runtime/panic.go:1038 +0x215
errors.As({0x7f81a5f5a1d8, 0xc001addf38}, {0xc000e24080, 0xc0016151b0})
	errors/wrap.go:88 +0x3d2
reflect.Value.call({0x2f04a80, 0x3b2dd48, 0x416725}, {0x3633d43, 0x4}, {0xc000e118c0, 0x2, 0x7f81a5b75748})
	reflect/value.go:556 +0x845
reflect.Value.Call({0x2f04a80, 0x3b2dd48, 0x451a72}, {0xc000e118c0, 0x2, 0x2})
	reflect/value.go:339 +0xc5
github.com/traefik/yaegi/interp.callBin.func1({0x2f04a80, 0x3b2dd48, 0x8}, {0xc000e118c0, 0xc001870c98, 0x4a7eb3})
	github.com/traefik/yaegi@v0.11.2/interp/run.go:1472 +0x28
github.com/traefik/yaegi/interp.callBin.func7(0xc0004c18c0)
	github.com/traefik/yaegi@v0.11.2/interp/run.go:1581 +0x163
github.com/traefik/yaegi/interp.runCfg(0xc000cddb00, 0xc0004c18c0, 0x39, 0x3528580)
	github.com/traefik/yaegi@v0.11.2/interp/run.go:201 +0x2ac
github.com/traefik/yaegi/interp.genFunctionWrapper.func2.1({0xc000e11020, 0x2, 0x4})
	github.com/traefik/yaegi@v0.11.2/interp/run.go:1023 +0x4a5
github.com/traefik/yaegi/stdlib._net_http_Handler.ServeHTTP(...)
	github.com/traefik/yaegi@v0.11.2/stdlib/go1_17_net_http.go:288
github.com/traefik/traefik/v2/pkg/middlewares/accesslog.(*FieldHandler).ServeHTTP(0xc0006cc480, {0x3fa6148, 0xc001614a70}, 0x2f9c120)
	github.com/traefik/traefik/v2/pkg/middlewares/accesslog/field_middleware.go:29 +0x122
github.com/gorilla/mux.(*Router).ServeHTTP(0xc00081a4e0, {0x3fa6148, 0xc001614a70}, 0xc0006c9700)
	github.com/gorilla/mux@v1.8.0/mux.go:141 +0x24c
github.com/traefik/traefik/v2/pkg/middlewares/recovery.(*recovery).ServeHTTP(0xfffffec0fffffeb, {0x3fa6148, 0xc001614a70}, 0x2df1120)
	github.com/traefik/traefik/v2/pkg/middlewares/recovery/recovery.go:32 +0x82
github.com/traefik/traefik/v2/pkg/middlewares/accesslog.(*FieldHandler).ServeHTTP(0xc0006cc780, {0x3fa6148, 0xc001614a70}, 0x7fb000000f9)
	github.com/traefik/traefik/v2/pkg/middlewares/accesslog/field_middleware.go:29 +0x122
github.com/traefik/traefik/v2/pkg/middlewares/snicheck.SNICheck.ServeHTTP({{0x3f65680, 0xc0006cc780}, 0xc0008404e0}, {0x3fa6148, 0xc001614a70}, 0xc0006c9700)
	github.com/traefik/traefik/v2/pkg/middlewares/snicheck/snicheck.go:49 +0x189
github.com/traefik/traefik/v2/pkg/middlewares.(*HTTPHandlerSwitcher).ServeHTTP(0x4101a7, {0x3fa6148, 0xc001614a70}, 0x3f1c701)
	github.com/traefik/traefik/v2/pkg/middlewares/handler_switcher.go:23 +0x62
github.com/traefik/traefik/v2/pkg/middlewares/requestdecorator.(*RequestDecorator).ServeHTTP(0xc00000e7d0, {0x3fa6148, 0xc001614a70}, 0xc0006c9600, 0xc0006a7c00)
	github.com/traefik/traefik/v2/pkg/middlewares/requestdecorator/request_decorator.go:47 +0x57b
github.com/traefik/traefik/v2/pkg/middlewares/requestdecorator.WrapHandler.func1.1({0x3fa6148, 0xc001614a70}, 0xc000e10e40)
	github.com/traefik/traefik/v2/pkg/middlewares/requestdecorator/request_decorator.go:84 +0x68
net/http.HandlerFunc.ServeHTTP(0xc000126370, {0x3fa6148, 0xc001614a70}, 0x9)
	net/http/server.go:2047 +0x2f
github.com/traefik/traefik/v2/pkg/middlewares/forwardedheaders.(*XForwarded).ServeHTTP(0xc000126370, {0x3fa6148, 0xc001614a70}, 0xc0006c9600)
	github.com/traefik/traefik/v2/pkg/middlewares/forwardedheaders/forwarded_header.go:189 +0xca
net/http.serverHandler.ServeHTTP({0x0}, {0x3fa6148, 0xc001614a70}, 0xc0006c9600)
	net/http/server.go:2879 +0x43b
net/http.initALPNRequest.ServeHTTP({{0x3fcd100, 0xc000e107e0}, 0xc00146f500, {0xc00054e540}}, {0x3fa6148, 0xc001614a70}, 0xc0006c9600)
	net/http/server.go:3480 +0x245
net/http.(*http2serverConn).runHandler(0x100000000000000, 0xc926a0, 0x0, 0x0)
	net/http/h2_bundle.go:5849 +0x78
created by net/http.(*http2serverConn).processHeaders
	net/http/h2_bundle.go:5579 +0x510
" middlewareName=traefik-internal-recovery middlewareType=Recovery
```

**Comment by nscuro** [comment]
Thanks for reporting! Indeed I can reproduce this, but I've *no idea* why it's happening. 

I've tried a few different things for a while and am now at a point where requests are answered with the configured status code again. However, for each request, traefik is now logging:

```
time="2022-04-03T18:28:39Z" level=error msg="2022/04/03 18:28:39 geoblock@file: <nil>" module=github.com/nscuro/traefik-plugin-geoblock plugin=plugin-geoblock
time="2022-04-03T18:28:40Z" level=error msg="2022/04/03 18:28:40 geoblock@file: <nil>" plugin=plugin-geoblock module=github.com/nscuro/traefik-plugin-geoblock
```

I think something's messed up in traefik, and I need more time to investigate this...

The panic in the stacktrace you posted is actually triggered [here](https://github.com/nscuro/traefik-plugin-geoblock/blob/v0.4.0/plugin.go#L95). Looks like `errors.As` has a bug in yaegi right now.

**Comment by nscuro** [comment]
This is now fixed in v0.5.0. I also added some more context to the log message when a request is blocked, e.g.

```
time="2022-04-09T13:30:01Z" level=error msg="2022/04/09 13:30:01 geoblock@file: [app.example.com GET /login] blocked request from BE" plugin=plugin-geoblock module=github.com/nscuro/traefik-plugin-geoblock
```
